src/Component/Controller/CurrentRequestTrait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Sonata\Component\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+/**
+ * NEXT_MAJOR: Remove this trait
+ *
+ * @deprecated Trait to be removed in 3.x
+ *
+ * @author Anton Zlotnikov <exp.razor@gmail.com>
+ */
+trait CurrentRequestTrait
```

Comment:
👎 for this trait, because it could be used by other people => we couldn't drop this on a stable branch